contracts/proxy/contract.go

```diff
@@ -42,8 +74,67 @@ func Update(nefFile, manifest []byte, data any) {
 
 // Verify checks whether carrier transaction contains either (2/3N + 1) or
 // (N/2 + 1) valid multi-signature of the NeoFS Alphabet.
+// Container contract's `SubmitObjectPut` is an exception, Alphabet signature
+// is not required, contract's `VerifyPlacementSignatures` is called instead.
 func Verify() bool {
-	return common.ContainsAlphabetWitness()
+	cnrH := storage.Get(storage.GetReadOnlyContext(), containerContractKey).(interop.Hash160)
+	tx := runtime.GetScriptContainer()
+	if string(tx.Script[len(tx.Script)-5:]) != string([]byte{65, 98, 125, 91, 82}) { // SystemContractCall
+		return common.ContainsAlphabetWitness()
+	}
+	scriptNoSyscall := tx.Script[:len(tx.Script)-5]
+	scriptLen := len(scriptNoSyscall)
+	if !cnrH.Equals(scriptNoSyscall[scriptLen-20:]) {
+		return common.ContainsAlphabetWitness()
+	}
+
+	scriptNoContract := scriptNoSyscall[:scriptLen-20-1-1]
```

Comment:
PUSHDATA1 + length? These bytes need to be checked as well.

Reply:
ok, but still: https://github.com/nspcc-dev/neofs-contract/pull/494#discussion_r2129432493. i cannot understand where i am wrong